types/wlr_seat.c

```diff
@@ -15,29 +20,22 @@ static void wl_pointer_set_cursor(struct wl_client *client,
 	wlr_log(L_DEBUG, "TODO: wl_pointer_set_cursor");
 }
 
+static const struct wl_pointer_interface wl_pointer_impl = {
+	.set_cursor = wl_pointer_set_cursor,
+	.release = resource_destroy
+};
+
 static void wl_pointer_destroy(struct wl_resource *resource) {
 	struct wlr_seat_handle *handle = wl_resource_get_user_data(resource);
 	if (handle->pointer) {
-		wl_resource_destroy(handle->pointer);
 		handle->pointer = NULL;
 	}
 }
 
-static void wl_pointer_release(struct wl_client *client,
-		struct wl_resource *resource) {
-	wl_pointer_destroy(resource);
-}
-
-struct wl_pointer_interface wl_pointer_impl = {
-	.set_cursor = wl_pointer_set_cursor,
-	.release = wl_pointer_release
-};
-
 static void wl_seat_get_pointer(struct wl_client *client,
 		struct wl_resource *_handle, uint32_t id) {
 	struct wlr_seat_handle *handle = wl_resource_get_user_data(_handle);
-	if (!(handle->wlr_seat->capabilities & WL_SEAT_CAPABILITY_KEYBOARD)) {
-		// TODO: Show error?
```

Comment:
There is no error code for this and making up own codes isn't a thing i guess. If i did not oversee something weston does not do it either.